### PR TITLE
Support Consul namespaces in consul-k8s

### PR DESCRIPTION
**Pull request by adilyse**
Placeholder PR for namespaces feature branch.

### CODE REVIEW

**Comment by ishustava** [comment]
FYI: we'll need #202 before we merge this. Otherwise, envoy fails to start because the bootstrap config has values that aren't supported in our default version.

File: subcommand/sync-catalog/command_ent_test.go [hunk 1]
```diff
@@ -477,6 +553,175 @@ func TestRun_ToConsulChangingNamespaceFlags(t *testing.T) {
...
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "non-default",
+				},
+			})
+			require.NoError(tt, err)
+			_, err = k8s.CoreV1().Services("non-default").Create(lbService("non-default", "1.1.1.1"))
+			require.NoError(tt, err)
+
+			// Set up consul server
+			a, err := testutil.NewTestServerConfigT(t, func(client *testutil.TestServerConfig) {
```
Comment (lkysow) [inline subcommand/sync-catalog/command_ent_test.go]:
```suggestion
			a, err := testutil.NewTestServerConfigT(tt, func(client *testutil.TestServerConfig) {
```

File: go.mod [hunk 2]
```diff
@@ -7,15 +7,16 @@ require (
...
 	github.com/docker/go-connections v0.4.0 // indirect
 	github.com/elazarl/go-bindata-assetfs v1.0.0 // indirect
 	github.com/gogo/protobuf v1.3.1 // indirect
 	github.com/golang/groupcache v0.0.0-20180513044358-24b0969c4cb7 // indirect
 	github.com/googleapis/gnostic v0.3.1 // indirect
 	github.com/gregjones/httpcache v0.0.0-20190611155906-901d90724c79 // indirect
-	github.com/hashicorp/consul v1.6.2
-	github.com/hashicorp/consul/api v1.3.0
-	github.com/hashicorp/consul/sdk v0.3.0
+	github.com/hashicorp/consul v1.7.0-beta3.0.20200125022210-1f6bf7edfb7e
```
Comment (lkysow) [inline go.mod]: We'll use the right tag in a subsequent PR.

**Comment by adilyse** [comment]
Personally I find the `Files changed: 36` even more shocking 🙃

**Pull request merged**